js/krakenfutures.js

```diff
@@ -0,0 +1,1433 @@
+            type = this.parseOrderType (type);
+        }
+        let symbol = undefined;
+        if (symbolId !== undefined) {
+            if (symbolId in this.markets_by_id) {
+                market = this.markets_by_id[symbolId];
+            } else {
+                market = undefined;
+                symbol = symbolId;
+            }
+        }
+        if (market !== undefined) {
+            symbol = market['symbol'];
+        }
+        let cost = undefined;
+        if ((amount !== undefined) && (price !== undefined) && (market !== undefined)) {
+            if (market['linear']) {
+                cost = amount * price; // in quote
+            } else {
+                cost = amount / price; // in base
+            }
+            cost *= market['lotSize'];
+        }
+        const fee = undefined;
+        let takerOrMaker = undefined;
+        const fillType = this.safeString (trade, 'fillType');
+        if (fillType !== undefined) {
+            if (fillType.indexOf ('taker') >= 0) {
+                takerOrMaker = 'taker';
+            } else if (fillType.indexOf ('maker') >= 0) {
+                takerOrMaker = 'maker';
+            }
+        }
+        return {
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': id,
+            'order': order,
+            'type': type,
+            'takerOrMaker': takerOrMaker,
+            'side': side,
+            'price': price,
+            'cost': cost,
+            'amount': amount,
+            'fee': fee,
+            'info': trade,
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        /**
+            @param {string} symbol: CCXT market symbol
+            @param {string} type: One of 'limit', 'market', 'post' (post-only limit order), 'take_profit' (take_profit order)
+            @param {string} side: buy or sell
+            @param {integer} amount: Contract quantity
+            @param {float} price: Limit order price
+            @param {float} params.stopPrice: The stop price associated with a stop or take profit order. Required if orderType is stp or take_profit. Must not have more than 2 decimal places. Note that for stop orders, limitPrice denotes the worst price at which the stop or take_profit order can get filled at. If no limitPrice is provided the stop or take_profit order will trigger a market order.
+            @param {string} params.triggerSignal: If placing a stp or take_profit, the signal used for trigger. One of: 'mark', 'index', 'last' (market price)
+            @param {string} params.cliOrdId: UUID - The order identity that is specified from the user. It must be globally unique.
+            @param {boolean} params.reduceOnly: Set as true if you wish the order to only reduce an existing position. Any order which increases an existing position will be rejected. Default false.
```

Comment:
@frosty00 @kroitor @ttodua @Dan-krm @carlosmiei 

So I think that making use of docstrings would actually be pretty helpful for the sake of specifying exchange specific parameters. It makes them easy to find for devs, and we can create our own exchange specific documentation that is more condensed that the exchanges documentation

For example, Kraken Futures order types are

- `lmt` for a limit order
- `post` for a post-only limit order
- `mkt` for an immediate-or-cancel order with 1% price protection
- `stp` for a stop order
- `take_profit` for a take profit order
- `ioc` for an immediate-or-cancel order

and I set the type param doc as

`@param {string} type: One of 'limit', 'market', 'post' (post-only limit order), 'take_profit' (take_profit order)`

because we have a unified way of doing `timeInForce` and `stp`, but we don't have a unified way for `post-only` or `take-profit`

---------------

I think it would be good if we got in the habit of doing this because we could create exchange specific documentation that would be a lot more clear than the exchange documentation

Reply:
@ttodua @kroitor We already copy a lot of the exchanges docs indirectly throw param checking, in that you need to understand the exchanges docs to know why we're doing a lot of the stuff. In my opinion docstrings aren't really any extra effort, they only take a few minutes, and they will be helpful for development, so they'll save time also.